batavia/types/Float.js

```diff
@@ -314,12 +317,25 @@ Float.prototype.__mul__ = function(other) {
 Float.prototype.__mod__ = function(other) {
     var types = require('../types')
 
-    /* TODO: Fix case for -0.0, which is coming out 0.0 */
     if (types.isinstance(other, types.Int)) {
         if (other.val.isZero()) {
             throw new exceptions.ZeroDivisionError.$pyclass('float modulo')
         } else {
-            return new Float(python_modulo(this.valueOf(), parseFloat(other.val)))
+            var thisNum = this.valueOf()
+            var otherNum = parseFloat(other.val)
+            var result = new Float(python_modulo(thisNum, otherNum))
+            if (otherNum > MAX_FLOAT || otherNum < MIN_FLOAT || result.toString() === 'nan' || result.toString() === 'inf' || result.toString() === '-inf'){
+                throw new exceptions.OverflowError.$pyclass(
+                    'int too large to convert to float'
+                )
+            }
+            if (otherNum > thisNum && thisNum > 0 || thisNum > otherNum && thisNum < 0 || thisNum === 0) {
+                return new Float(thisNum)
+            }
+            if (result.valueOf() === 0 && (thisNum%otherNum)+otherNum === otherNum){
+                return new Float(otherNum)
+            }
```

Comment:
At column 36: [(no-mixed-operators) Unexpected mix of '&&' and '||'.](http://eslint.org/docs/rules/no-mixed-operators)